### PR TITLE
Raise an error when do level does nothing

### PR DESCRIPTION
**Pull request by enadeau**

### CODE REVIEW

**Comment by enadeau** [comment]
It's not the auto search that's broken but the expansion in the creation of the specification. I'll fix that

File: comb_spec_searcher/specification.py [hunk 1]
```diff
@@ -139,13 +139,19 @@ def _expand_verified_comb_classes(
...
+                except NoMoreClassesToExpandError:
+                    new_rules = css.ruledb.get_smallest_specification(css.start_label)
+                    done_expanding = True
                 try:
                     new_rules = css.ruledb.get_smallest_specification(css.start_label)
                 except SpecificationNotFound:
-                    continue
+                    pass
+                else:
+                    done_expanding = True
```
Comment (christianbean) [inline comb_spec_searcher/specification.py:154]: Can we simplify the logic too:
```python
            while True:
                try:
                    css.do_level()
                except NoMoreClassesToExpandError:
                    new_rules = css.ruledb.get_smallest_specification(css.start_label)
                    break
                try:
                    new_rules = css.ruledb.get_smallest_specification(css.start_label)
                    break
                except SpecificationNotFound:
                    pass
```

Comment (enadeau) [reply comb_spec_searcher/specification.py:154]: Yes looks better

Comment (enadeau) [reply comb_spec_searcher/specification.py]: Done.

**Pull request merged**